runtime/intern.c

```diff
@@ -901,10 +998,16 @@ CAMLprim value caml_marshal_data_size(value buff, value ofs)
                   "object too large to be read back on a 32-bit platform");
 #endif
     break;
+  case Intext_magic_number_compressed:
+    header_len = read8u(s) & 0x3F;
+    if (readvlq(s, &data_len) != 0)
+      caml_failwith("Marshal.data_size: "
+                    "object too large to be read back on a 32-bit platform");
```

Comment:
minor: if there is a corrupted stream this message might be printed on a 64-bit platform too. Elsewhere the error message says 'this platform', which is probably better. (the messages which say '32-bit platform' are usually related to the use of Compat_32).

Reply:
Agreed.  Done in ed15d5b45a